doc/reference/misc/timeutil.rst

```diff
@@ -0,0 +1,289 @@
+EPOCH is converted to ``struct tm`` representing calendar time with
+`gmtime()
+<https://pubs.opengroup.org/onlinepubs/9699919799/functions/gmtime.html>`__.
+Sub-second timestamps like ``struct timespec`` can also use this to
+produce the calendar time representation and deal with sub-second
+offsets separately.
+
+The inverse transformation is not standardized: APIs like ``mktime()``
+expect information about time zones.  Zephyr module provides this
+transformation with :c:func:`timeutil_timegm` and
+:c:func:`timeutil_timegm64`.
+
+.. doxygengroup:: timeutil_repr_apis
+   :project: Zephyr
+
+.. _timeutil_sync:
+
+Time Scale Synchronization
+==========================
+
+There are several factors that affect synchronizing time scales:
+
+* The rate of discrete instant representation change.  For example
+  Zephyr uptime is tracked in ticks which advance at
+  :option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC` Hertz, while an external time
+  source may provide data in whole or fractional seconds
+  (e.g. microseconds).
+* The absolute offset required to align the two scales at a single
+  instant.
+* The relative error between observable instants in each scale, requred
+  to align multiple instants consistently.  For example a reference
+  clock that's conditioned by a 1-pulse-per-second GPS signal will be
+  much more accurate than a Zephyr system clock driven by a RC
+  oscillator with a +/- 250 ppm error.
+
+Synchronization or alignment between time scales is done with a
+multi-step process:
+
+* An instant in a time scale is represented by an (unsigned) 64-bit
+  integer, assumed to advance at a fixed nominal rate.
+* :c:struct:`timeutil_sync_config` records the nominal rates of a
+  reference time scale/source (e.g. TAI) and a local time source
+  (e.g. :c:func:`k_uptime_ticks`).
+* :c:struct:`timeutil_sync_instant` records the representation of a
+  single instant in both the reference and local time scales.
+* :c:struct:`timeutil_sync_state` provides storage for an initial
+  instant, a recently received second observation, and a skew that can
+  adjust for relative errors in the actual rate of each time scale.
+* :c:func:`timeutil_sync_ref_from_local()` and
+  :c:func:`timeutil_sync_local_from_ref()` convert instants in one time
+  scale to another taking into account skew that can be estimated from
+  the two instances stored in the state structure by
+  :c:func:`timeutil_sync_estimate_skew`.
+
+.. doxygengroup:: timeutil_sync_apis
+   :project: Zephyr
+
+.. _timeutil_realtime:
+
+System Time Conversion
+======================
+
+The :option:`CONFIG_TIMEUTIL_REALTIME` flag enables API that allows
+zephyr to provide the current civil ("real") time that advances aligned
+to :ref:`kernel_timing_uptime`.
+
+Something in the application must invoke
+:c:func:`timeutil_realtime_set()` at least once with a reference
+timestamp and the :c:func:`k_uptime_ticks()` value corresponding to the
```

Comment:
So essentially the 'seed' is created at the time `timeutil_realtime_set()` is called. Is there any additional information on when/where this should be called for optimal use of the API (I can imagine, the earlier the better).